cfme/ansible_tower/jobs.py

```diff
@@ -51,14 +61,79 @@ def is_displayed(self):
         )
 
 
+class AnsibleTowerJobsDetailsView(TowerJobsView):
+    @View.nested
+    class entities(View):  # noqa
+        """ Represents details page when it's switched to Summary/Table view """
+        properties = SummaryTable(title="Properties")
+        relationships = SummaryTable(title="Relationships")
+        smart_management = SummaryTable(title="Smart Management")
+
+    @property
+    def is_displayed(self):
+        """Is this view being displayed?"""
+        return self.title.text == '{} (Summary)'.format(self.context['object'].template_name)
+
+
 @attr.s
-class TowerJobs(BaseEntity):
-    pass
+class TowerJob(BaseEntity):
+    template_name = attr.ib()
+
+    @property
+    def status(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of("Status")
+
+    def delete(self):
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this Job', handle_alert=True)
+        view.flash.assert_no_error()
+
+    def is_job_successful(self):
+        return True if self.status == "successful" else False
+
+    def wait_for_completion(self, num_sec=1200, delay=10):
```

Comment:
This is actually waiting for successful completion, because its relying on `is_job_successful`.   Either it needs to renamed, or re-written to look for multiple states. It would be good to enumerate those possible states on the entity class so that they can be referenced consistently.